### PR TITLE
Implement icon-button in typescript

### LINKED ISSUE #95: Publish mwc-icon-button

### PR DESCRIPTION
**Pull request by azakus**
- icon-toggle is now icon-button, matching MDC
- If `offLabel` is not defined, button is always "on" and does not
toggle

Fixes #95

### CODE REVIEW

**Comment by azakus** [comment]
This element is the first attempt at implementing #140

File: packages/icon-button/src/icon-button-base.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  updated() {
+    if (this.offIcon === '') {
+      this.on = true;
+    }
+  }
+
+  render() {
+    return html`
+      ${this.renderStyle()}
```
Comment (sorvell) [inline packages/icon-button/src/icon-button-base.ts]: Let's make sure there's an abstract `renderStyle`

File: packages/icon-button/src/icon-button-base.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        }
+        this.dispatchEvent(new CustomEvent('MDCIconButtonToggle:change', {detail: evtData, bubbles: true}));
+      }
+    }
+  }
+
+  protected handleClick() {
+    if (this.offIcon !== '') {
+      this.on = !this.on;
+      this.mdcFoundation.handleClick();
```
Comment (sorvell) [inline packages/icon-button/src/icon-button-base.ts]: Are we sure that calling `handleClick` on the foundation and then `toggle` in the observer works? It seems like that could toggle the value 2x. If it doesn't, why not?

File: packages/icon-button/src/icon-button-base.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  label = '';
+
+  @property({type: Boolean, reflect: true})
+  disabled = false;
+
+  @property({type: String})
+  icon = '';
+
+  @property({type: String})
+  offIcon = '';
```
Comment (sorvell) [inline packages/icon-button/src/icon-button-base.ts]: We discussed making the `offIcon` not optional, but if it is not, how is the behavior different? Would we throw if there's no `offIcon`? That would be strange.

File: packages/icon-button/src/icon-button-base.ts (hunk 1 above)
Comment (azakus) [reply packages/icon-button/src/icon-button-base.ts]: This is already abstract in `BaseElement`

File: packages/icon-button/src/icon-button-base.ts (hunk 2 above)
Comment (azakus) [reply packages/icon-button/src/icon-button-base.ts]: toggle just calls the adapter `addClass` twice, which uses `classList.add()`, and that should be a no-op to add the same class twice.

File: packages/icon-button/src/icon-button-base.ts (hunk 3 above)
Comment (azakus) [reply packages/icon-button/src/icon-button-base.ts]: If there is no `offIcon`, the state never toggles and we do not emit an event.

**Pull request merged**